src/PostgREST/DbStructure.hs

```diff
@@ -617,69 +591,36 @@ allPrimaryKeys tabs =
   H.Statement sql HE.noParams (decodePks tabs) True
  where
   sql = [q|
-    /*
     -- CTE to replace information_schema.table_constraints to remove owner limit
-    */
     WITH tc AS (
-        SELECT current_database()::information_schema.sql_identifier AS constraint_catalog,
-            nc.nspname::information_schema.sql_identifier AS constraint_schema,
-            c.conname::information_schema.sql_identifier AS constraint_name,
-            current_database()::information_schema.sql_identifier AS table_catalog,
-            nr.nspname::information_schema.sql_identifier AS table_schema,
-            r.relname::information_schema.sql_identifier AS table_name,
-                CASE c.contype
-                    WHEN 'c'::"char" THEN 'CHECK'::text
-                    WHEN 'f'::"char" THEN 'FOREIGN KEY'::text
-                    WHEN 'p'::"char" THEN 'PRIMARY KEY'::text
-                    WHEN 'u'::"char" THEN 'UNIQUE'::text
-                    ELSE NULL::text
-                END::information_schema.character_data AS constraint_type,
```

Comment:
This was only used to filter for 'PRIMARY KEY' in the outside query, so its much simpler to filter directly here in the CTE

Reply:
Yes, and since you filtered the results by pk constraint(`c.contype = 'p'`) the other condition for omitting trigger and exclusion constraints is not needed(`(c.contype <> ALL (ARRAY['t', 'x']))` ).

It's much more clear now!